src/hotspot/cpu/x86/x86.ad

```diff
@@ -8269,6 +8598,569 @@ instruct vmask_first_or_last_true_avx(rRegI dst, vec mask, rRegL tmp, vec xtmp,
+  %}
+  ins_pipe( pipe_slow );
+%}
+
+instruct vmul_mem_masked(vec dst, memory src2, kReg mask) %{
+  predicate(UseAVX > 2);
+  match(Set dst (MulVS (Binary dst src2) mask));
+  match(Set dst (MulVI (Binary dst src2) mask));
+  match(Set dst (MulVL (Binary dst src2) mask));
+  match(Set dst (MulVF (Binary dst src2) mask));
+  match(Set dst (MulVD (Binary dst src2) mask));
+  format %{ "vpmul_masked $dst, $dst, $src2\t! mul masked operation" %}
+  ins_encode %{
+    int vlen_enc = vector_length_encoding(this);
+    BasicType bt = vector_element_basic_type(this);
+    int opc = this->ideal_Opcode();
+    __ evmasked_op(opc, bt, $mask$$KRegister, $dst$$XMMRegister,
+                   $dst$$XMMRegister, $src2$$Address, true, vlen_enc);
+  %}
+  ins_pipe( pipe_slow );
+%}
+
+instruct vdiv_reg_masked(vec dst, vec src2, kReg mask) %{
+  predicate(UseAVX > 2);
+  match(Set dst (DivVF (Binary dst src2) mask));
+  match(Set dst (DivVD (Binary dst src2) mask));
+  format %{ "vpdiv_masked $dst, $dst, $src2\t! div masked operation" %}
+  ins_encode %{
+    int vlen_enc = vector_length_encoding(this);
+    BasicType bt = vector_element_basic_type(this);
+    int opc = this->ideal_Opcode();
+    __ evmasked_op(opc, bt, $mask$$KRegister, $dst$$XMMRegister,
+                   $dst$$XMMRegister, $src2$$XMMRegister, true, vlen_enc);
+  %}
+  ins_pipe( pipe_slow );
+%}
+
+instruct vdiv_mem_masked(vec dst, memory src2, kReg mask) %{
+  predicate(UseAVX > 2);
+  match(Set dst (DivVF (Binary dst src2) mask));
+  match(Set dst (DivVD (Binary dst src2) mask));
+  format %{ "vpdiv_masked $dst, $dst, $src2\t! div masked operation" %}
+  ins_encode %{
+    int vlen_enc = vector_length_encoding(this);
+    BasicType bt = vector_element_basic_type(this);
+    int opc = this->ideal_Opcode();
+    __ evmasked_op(opc, bt, $mask$$KRegister, $dst$$XMMRegister,
+                   $dst$$XMMRegister, $src2$$Address, true, vlen_enc);
+  %}
+  ins_pipe( pipe_slow );
+%}
+
+instruct vlshift_reg_masked(vec dst, vec src2, kReg mask) %{
+  predicate(UseAVX > 2);
+  match(Set dst (LShiftVS (Binary dst src2) mask));
+  match(Set dst (LShiftVI (Binary dst src2) mask));
+  match(Set dst (LShiftVL (Binary dst src2) mask));
+  format %{ "vplshift_masked $dst, $dst, $src2\t! lshift masked operation" %}
+  ins_encode %{
+    int vlen_enc = vector_length_encoding(this);
+    BasicType bt = vector_element_basic_type(this);
+    int opc = this->ideal_Opcode();
+    bool is_varshift = !VectorNode::is_vshift_cnt_opcode(in(2)->isa_Mach()->ideal_Opcode());
+    __ evmasked_op(opc, bt, $mask$$KRegister, $dst$$XMMRegister,
+                   $dst$$XMMRegister, $src2$$XMMRegister, true, vlen_enc, is_varshift);
+  %}
+  ins_pipe( pipe_slow );
+%}
+
+instruct vlshift_mem_masked(vec dst, memory src2, kReg mask) %{
+  predicate(UseAVX > 2);
+  match(Set dst (LShiftVS (Binary dst src2) mask));
+  match(Set dst (LShiftVI (Binary dst src2) mask));
+  match(Set dst (LShiftVL (Binary dst src2) mask));
```

Comment:
Could we add the match rules for RShift, URShift here?
The ins_encode part it the same for all of these.

Reply:
Same as above